### PR TITLE
feat: upgrade general versions of all stripe dependencies and others

### LINKED ISSUE #27: Is this module still maintained ?
Hello,

Was looking around to implement Stripe with Nuxt and checking out this module. Seems like this was not updated for the last 6 months. 
During those 6 months there were several updates on the SDKs and APIs from Stripe.

Is there a benefit to use this instead of implementing those by hand with the updated versions?

Will this ever be updated again?

Have a wonderful day, Colin.

### PR DESCRIPTION
**Pull request by flozero**
close #27 

## Types of changes

- [x] New feature (a non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Upgrade major versions of stripejs and node make sure everything is working nicely

## Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes (if not applicable, please state why)

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1 @@
+yarn test
```
Comment (fuentesloic) [inline .husky/pre-commit:1]: 🥇

File: README.md [hunk 2]
```diff
@@ -142,22 +138,13 @@ yarn release
 npm run release
 ```
 
-## Nuxt 2
```
Comment (fuentesloic) [inline README.md:145]: 🔪

File: src/module.ts [hunk 3]
```diff
@@ -30,9 +31,7 @@ export default defineNuxtModule<ModuleOptions>({
   defaults: {
       server: {
         key: null,
-        options: {
-          apiVersion: '2022-11-15'
```
Comment (fuentesloic) [inline src/module.ts:34]: 👍

File: playground/nuxt.config.ts [hunk 4]
```diff
@@ -2,10 +2,10 @@ export default defineNuxtConfig({
   modules: ['../src/module'],
   stripe: {
     client: {
-      key: 'pk_test_123'
+      key: 'pk_test_51MdmAOBcGKkz735sAoGNx8c08M8UNcXe53twmQuWBm3zxlicpTGbeoi3qGiMayPcNUK6EOBe7f1kUezJvdSgwco800ONPEXqs0'
```
Comment (fuentesloic) [inline playground/nuxt.config.ts:5]: is the stripe test key in their example?

File: src/runtime/composables/useClientStripe.ts [hunk 5]
```diff
@@ -1,39 +1,41 @@
...
+  const isLoading = useState('stripe-client-loading', () => false)
+
+  async function _loadStripe() {
+    if (stripe.value){
+      return stripe.value
+    }
+
+    isLoading.value = true
+
+    if (!nuxtApp.$config.public.stripe.key) console.warn("no key given for client service")
```
Comment (fuentesloic) [inline src/runtime/composables/useClientStripe.ts:24]:
```suggestion
    if (!nuxtApp.$config.public.stripe.key) console.warn("no key given for Stripe client service")
```

File: src/runtime/composables/useClientStripe.ts [hunk 6]
```diff
@@ -1,39 +1,41 @@
...
+    return await loadStripe(
+      nuxtApp.$config.public.stripe.key,
+      nuxtApp.$config.public.stripe.options
+    )
+  }
+
+  onMounted(async () => {
+    if (!isLoading.value) {
+      const _stripe = await _loadStripe()
+      stripe.value = _stripe
```
Comment (fuentesloic) [inline src/runtime/composables/useClientStripe.ts:35]: NIT at this point
```
stripe.value = await _loadStripe()
```

File: playground/nuxt.config.ts (hunk 4 above)
Comment (fuentesloic) [reply playground/nuxt.config.ts:5]: https://github.com/fuentesloic/nuxt-stripe/pull/15/files#r1244514872

File: src/module.ts (hunk 3 above)
Comment (flozero) [reply src/module.ts:34]: Just for info I removed it as by default stripe will take the latest. But we can override it ofc

**Review by fuentesloic — APPROVED** [review]
let's go! I will run the playground soon as QA step and then merge it 🚀

**Pull request merged**